src/hotspot/share/oops/instanceKlass.cpp

```diff
@@ -3185,6 +3185,33 @@ Method* InstanceKlass::method_at_itable(Klass* holder, int index, TRAPS) {
   return m;
 }
 
+int InstanceKlass::vtable_index_of_interface_method(Method* intf_method) {
+  assert(this->is_linked(), "required");
+  assert(intf_method->method_holder()->is_interface(), "not an interface method");
+  assert(this->is_subtype_of(intf_method->method_holder()), "interface not implemented");
+
+  int vtable_index = Method::invalid_vtable_index;
+  Symbol* name = intf_method->name();
+  Symbol* signature = intf_method->signature();
+
+  // First check in default method array
+  if (!intf_method->is_abstract() && this->default_methods() != NULL) {
+    int index = InstanceKlass::find_method_index(this->default_methods(),
```

Comment:
Yes, please, this looks a lot better!  Then linkResolver can deal with methodHandle consistently so it doesn't go away with redefinition, and this function is nice here with the other like functions.
It doesn't need InstanceKlass:: before find_method_index, or the this->'s.

Reply:
> It doesn't need InstanceKlass:: before find_method_index, or the this->'s.

Fixed. It looked clearer to me to explicitly refer to `this`.